match/tensor.py

```diff
@@ -0,0 +1,206 @@
+from __future__ import annotations
+
+from copy import deepcopy
+from logging import info
+
+from .tensordata import TensorData
+
+
+class Tensor(object):
+    def __init__(self, data: TensorData, children: tuple = ()) -> None:
+        """A matrix object that tracks computations for computing gradients."""
```

Comment:
We'll need to document the purpose of `children`, and the API may need to change to be more torch-like.